src/test/java/org/jenkins/plugins/lockableresources/LockStepTest.java

```diff
@@ -823,6 +823,31 @@ public void lockWithLabelAndLabeledResourceQuantity() throws Exception {
     assertNotNull(LockableResourcesManager.get().fromName("resource4"));
   }
 
+  @Test
+  //@Issue("JENKINS-XXXXX")
+  public void multipleLocksFillVariables() throws Exception {
+    LockableResourcesManager.get().createResourceWithLabel("resource1", "label1");
+    LockableResourcesManager.get().createResourceWithLabel("resource2", "label1");
+    WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+    p.setDefinition(
+      new CpsFlowDefinition(
+        "lock(label: 'label1', variable: 'someVar', quantity: 2) {\n"
+          + "  echo \"VAR IS ${env.someVar.split(',').sort()}\"\n"
+          + "  echo \"VAR0or1 IS $env.someVar0\"\n"
+          + "  echo \"VAR0or1 IS $env.someVar1\"\n"
+          + "  echo \"VAR2 IS $env.someVar2\"\n"
```

Comment:
This PR description mentions lack of `uppercaseProperties` compared to original PR it replays.
So just to clarify: here the variable names (prefix part at least) keeps the casing of the `lock(variable:...)` argument. Whatever the pipeline script author spelled, right?

Reply:
Correct, while it's a good practice to keep environment variable in uppercase, I think I'd leave it up to whoever decides to name their property to follow that or not. If we enforce upper-casing (or provide an option for it), I would argue that this should apply to the current `variable` or it starts getting weird. For example, what would be expected of

```
lock(label: 'label1', variable: 'someVar', `uppercaseProperties: true) {
  // should we use env.someVar or env.SOMEVAR ?
}
```

There are also many situations where the properties could end up as invalid environment variable names, such as using UTF8 characters, spaces, hyphens, etc. But since this wasn't handled for `variable` I chose to not handle it and leave it to the users to understand that if they use a emoji in the name of their variable it just... won't work